### PR TITLE
Maven support build extensions

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Fix loading Maven build extensions, see the commit messages for details.

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/maven-wagon-expected-output.yml [hunk 1]
```diff
@@ -29,21 +29,13 @@ project:
         message: "Could not get package information for dependency 'com.example:ftp-artifact:1.0.0':\
           \ ProjectBuildingException: Error resolving project artifact: Could not\
           \ transfer artifact com.example:ftp-artifact:pom:1.0.0 from/to ftp-repository\
-          \ (ftp://localhost:40123): Cannot access ftp://localhost:40123 with type\
-          \ default using the available connector factories: BasicRepositoryConnectorFactory\
+          \ (ftp://localhost:40123): Password not specified for repository ftp-repository\
```
Comment (fviernau) [inline analyzer/src/funTest/assets/projects/synthetic/maven-wagon-expected-output.yml:32]: commit message / nit: "There Maven can find the correct artifact in this
step and load it into the `ClassRealm` used when resolving dependencies
later."

Does "there" and "in this step" refer to the same thing?

**Review by fviernau — APPROVED** [review]

**Review by oheger-bosch — APPROVED** [review]
Nice! And a cool way of testing the fix.

File: analyzer/src/funTest/assets/projects/synthetic/maven-wagon-expected-output.yml (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/maven-wagon-expected-output.yml:32]: You are right, but I'd rather not wait for another CI round to fix the minor wording issue.

Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/maven-wagon-expected-output.yml:32]: Agree.

**Pull request merged**